searx/search/models.py

```diff
@@ -23,7 +24,13 @@ def __hash__(self):
         return hash((self.name, self.category))
 
 
-class SearchQuery:
+class BaseQuery(ABC):
+    """Contains properties among all query classes"""
+
+    query: str
```

Comment:
This is just a basic start for BaseQuery as I needed it for the answerers module, but the scope of the BaseQuery can increase as needed.